src/main/java/seedu/mark/ui/MainWindow.java

```diff
@@ -220,6 +222,16 @@ private void handleTabSwitchRequestIfAny(Tab tab) {
         }
     }
 
+    public Consumer<Url> getCurrentUrlChangeHandler() {
+        return new Consumer<Url>() {
```

Comment:
minor issue: can change to lambda for this?

Reply:
Fixed